src/arc.ts

```diff
@@ -229,9 +230,8 @@ export class Arc extends GraphNodeObserver {
       contractInfos = response.data.contractInfos as IContractInfo[]
       allContractInfos.push(...contractInfos)
       skip++
-    } while (contractInfos && (contractInfos as any).length > 0)
+    } while (contractInfos.length === MAX_BATCH_QUERY)
```

Comment:
for what it is worth, there is an edge case here of an empty array for the last contractInfos that would be pushed onto the collection.

Reply:
You're right, I will add a condition to push only if the array is not empty.
Thanks.